### PR TITLE
Clean logs and cycle files to limit size

### PR DESCRIPTION
**Pull request by RockClapps**
I added some code to regularly check if disk space is getting tight, and automatically clear old logs until more is available. I also added a `_0001` extension to the log names to be able to have multiple log files in one day, and move on to the next one when the current one is too big. 

I wasn't sure how to handle the issue of running out of values to name the file when you pass 9999, but that would require you to make 5TB of logs in one day so I am not sure it's worth making huge changes for

### CODE REVIEW

**Comment by RockClapps** [comment]
I also added some simple tests in the meantime to make sure it worked right and figured it wouldn't hurt to add them to the PR

File: tools.go [hunk 1]
```diff
@@ -2,15 +2,48 @@ package loge
...
+	pathToFile := filepath.Join(path, ret + fmt.Sprintf("%04d.log", fileNum))
+	fi, err := os.Stat(pathToFile)
+	if err == nil {
+		if fi.Size() > maxFileSize {
+			fileNum += 1
+		}
+	}
+
+	if fileNum > 9999 {
+		//idk man
```
Comment (spectorl) [inline tools.go:44]: Your intent here is how to number past 9999? I think we should come up with a better strategy. And obviously comments like "idk man" probably should use better terminology. :)

File: tools.go [hunk 2]
```diff
@@ -2,15 +2,48 @@ package loge
...
+	"os"
 	"path/filepath"
+	"sort"
+	"strconv"
 	"time"
 )
 
 const dateTimeStringLength = 27
 
+const maxFileSize = 5000000000
```
Comment (spectorl) [inline tools.go:14]: Maybe make the max size 10 MB? `10 * 1024 * 1024`?

File: tools.go [hunk 3]
```diff
@@ -2,15 +2,48 @@ package loge
...
+	fi, err := os.Stat(pathToFile)
+	if err == nil {
+		if fi.Size() > maxFileSize {
+			fileNum += 1
+		}
+	}
+
+	if fileNum > 9999 {
+		//idk man
+	}
```
Comment (spectorl) [inline tools.go:45]: Your intent here is how to number past 9999? I think we should come up with a better strategy. And obviously comments like "idk man" probably should use better terminology. :)

File: tools.go [hunk 4]
```diff
@@ -2,15 +2,48 @@ package loge
...
+	lastNum, _ := strconv.Atoi(lastFileName[9:13])
+	fileNum = lastNum
+
+	pathToFile := filepath.Join(path, ret + fmt.Sprintf("%04d.log", fileNum))
+	fi, err := os.Stat(pathToFile)
+	if err == nil {
+		if fi.Size() > maxFileSize {
+			fileNum += 1
+		}
+	}
```
Comment (spectorl) [inline tools.go:41]: We may want to think about how we'd like to handle if there is an error. If we fail to stat the file, we might want to pass the error back to the caller of `getLogName`

**Review by spectorl — CHANGES_REQUESTED** [review]

**Comment by spectorl** [comment]
Accidental merge. Please see comments here.

File: tools.go (hunk 3 above)
Comment (RockClapps) [reply tools.go:45]: The middleware just seems to loop back to 0 when file 9999 is full. I feel like something more should be done but if that works then I can just adjust it to do that

Comment (RockClapps) [reply tools.go:45]: Even then, middleware basically starts logging in the earliest number log file that isn't full. When all files are full, I think it throws a critical error

**Pull request merged**